oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -787,7 +787,28 @@ private static QueryBuilder fullTextQuery(String text, String fieldName, PlanRes
             // and could contain other parts like renditions, node name, etc
             return multiMatchQuery.field(fieldName);
         } else {
-            return simpleQueryStringQuery(text).field(fieldName).defaultOperator(Operator.AND);
+            // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
+            // simpleQueryStringQuery does not support leading wildcards whereas it's supported by default in queryStringQuery
+            // Not using queryStringQuery by default , since some functional cases break.
+            // simpleQueryStringQuery is less Strict, for instance searches for terms starting with / work, whereas
+            // with queryStringQuery, they throw an Exception (which ultimately results in an empty result set in oak),
+            // so using simpleQueryStringQuery by default would break certain functional cases.
+            // So only support this in case any term in the text String actually starts with *
+            // For example *hello or Hello *world
+            String[] textTerms = text.split(" ");
+            boolean allowLeadingWildCard = false;
+            for(String textTerm : textTerms) {
+                if (textTerm.startsWith("*")) {
+                    allowLeadingWildCard = true;
+                    break;
+                }
+            }
```

Comment:
You could use streams to be more concise:
```suggestion
            boolean allowLeadingWildCard = Arrays.stream(text.split(" "))
                    .anyMatch(textTerm -> textTerm.startsWith("*"));
```